homeassistant/components/telegram_bot/__init__.py

```diff
@@ -702,6 +719,49 @@ def edit_message(self, type_edit, chat_id=None, **kwargs):
                 timeout=params[ATTR_TIMEOUT],
                 parse_mode=params[ATTR_PARSER],
             )
+        if type_edit == SERVICE_EDIT_MEDIA:
+            media_type = kwargs.get(ATTR_MEDIA_TYPE)
+            file_content = load_data(
+                self.hass,
+                url=kwargs.get(ATTR_URL),
+                filepath=kwargs.get(ATTR_FILE),
+                username=kwargs.get(ATTR_USERNAME),
+                password=kwargs.get(ATTR_PASSWORD),
+                authentication=kwargs.get(ATTR_AUTHENTICATION),
+                verify_ssl=kwargs.get(ATTR_VERIFY_SSL),
+            )
+
+            if file_content:
+                if media_type == 'photo':
+                    return self._send_msg(
+                        self.bot.edit_message_media,
+                        "Error editing media to photo",
+                        params[ATTR_MESSAGE_TAG],
+                        chat_id=chat_id,
+                        message_id=message_id,
+                        inline_message_id=inline_message_id,
+                        media=InputMediaPhoto(media=file_content,
+                                              caption=kwargs.get(ATTR_CAPTION),
+                                              parse_mode=params[ATTR_PARSER]),
+                        reply_markup=params[ATTR_REPLYMARKUP],
+                    )
+
+                elif media_type == 'video':
+                    return self._send_msg(
+                        self.bot.edit_message_media,
+                        "Error editing media to video",
+                        params[ATTR_MESSAGE_TAG],
+                        chat_id=chat_id,
+                        message_id=message_id,
+                        inline_message_id=inline_message_id,
+                        media=InputMediaVideo(media=file_content,
+                                              caption=kwargs.get(ATTR_CAPTION),
+                                              parse_mode=params[ATTR_PARSER]),
+                        reply_markup=params[ATTR_REPLYMARKUP],
+                    )
```

Comment:
These branches are 99% the same – the only difference is the error message and the value of `media`. Sounds like that could be simplified.